rocAL/include/pipeline/node.h

```diff
@@ -23,10 +23,190 @@ THE SOFTWARE.
+            if (it != type_names.end()) {
+                enum_type_name = it->second;
+                values.push_back(static_cast<int>(val));
+            } else {
+                std::cout << "Type: Unknown" << arg_name << std::endl;
+            }
+        } else if constexpr (is_vector_type<T>::value) {
+            using ElementType = typename std::decay_t<T>::value_type;
+            auto it = type_names.find(typeid(ElementType));
+            if (it != type_names.end()) {
+                type_name = it->second;                
+                is_vector = true;
+                for (const auto& v : val) {
+                    // values.push_back(v);  // Store std::string as std::any
+                    values.push_back(static_cast<ElementType>(v));
+                }
+            } else {
+                std::cout << "Type: Unknown" << arg_name << std::endl;
+            }
+        } else {
+            auto it = type_names.find(typeid(std::decay_t<T>));
+            if (it != type_names.end()) {
+                type_name = it->second;
+
+                if constexpr (std::is_same<std::decay_t<T>, const char *>::value) {
+                    values.push_back(std::string(val));
+                } else {
+                    values.push_back(static_cast<std::decay_t<T>>(val));
+                }
+            } else {
+                std::cout << "Type: Unknown" << arg_name << std::endl;
+            }
+        }
+    }
+
+    // Used to store the feature key map
+    explicit inline Argument(const std::string& name, const std::map<std::string, std::string>&& val)
+        : arg_name(name) {
+        type_name = "map_string";
+        is_vector = true;
+        if (!val.empty()) {
+            for (const auto& pair : val) {
+                values.push_back(static_cast<std::string>(pair.first));  // Push key
+                values.push_back(static_cast<std::string>(pair.second)); // Push value
+            }
+        }
+    }
+
+    // Used to store the shared_ptr
+    template <typename T>
+    explicit inline Argument(const std::string& name, const std::shared_ptr<T>&& val)
+        : arg_name(name) {
+        type_name = "shared_ptr";
+
+        // For MetadataReader case store an empty value
+        // During deserialization the MetadataReader should be created and passed from the MasterGraph.
+        if (name == "meta_data_reader") {
+            values.push_back(static_cast<int>(0));
+        }
+    }
+
+    // Contructor to initialize the arguments of in-built data types
+    template <typename T>
+    explicit inline Argument(std::string name, std::string type, T val)
+        : arg_name(name), type_name(type) {
+        values.push_back(val);
+    }
+
+    // Constructor for vector type arguments
+    template <typename T>
+    explicit inline Argument(std::string name, std::string type, std::vector<T> &val)
+        : arg_name(name), type_name(type) {
+        is_vector = true;
+        for (const auto& v : val) {
```

Comment:
why do we need a for loop. Can't you assign one vector to other or use copy constructor